plugin/core/clients.py

```diff
@@ -36,6 +37,27 @@ def get_window_env(window: sublime.Window, config: ClientConfig) -> 'Tuple[List[
     return expanded_args, env
 
 
+def get_expanding_variables(window: sublime.Window) -> dict:
+    variables = window.extract_variables()
+    variables.update({
+        "home": os.path.expanduser('~'),
+        "temp_dir": tempfile.gettempdir(),
+    })
+
+    return variables
+
+
+def expand_variables_for_dict(window: sublime.Window,
+                              d: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
I don't think the old python3.3 from ST3 will accept this. Shouldn't this be in quotes?

`d: 'Dict[str, Any]') -> 'Dict[str, Any]'`

Reply:
Ah, yes!

Reply:
Why don't we use a real `typing` module?
https://github.com/packagecontrol/typing

Reply:
Interesting! I was not aware that ST dependency exists.

Reply:
About the dependency, do we really need `typing` to be importable for endusers?
We _could_ replace the try blocks with this (mypy sets MYPY to always True when checking):
```python
MYPY = False
if MYPY:
    from typing import Dict
```

Reply:
I was just curious why `typing` is not actually imported and using "quoted" type hintings in codes. For whatever concerned, you are in charge anyway. :smile: I am just lazy so I directly make it as a dependency in my plugins.